### PR TITLE
Add HTTPRoute request header docs

### PR DESCRIPTION
**Pull request by AliceProxy**
Adds a user guide for configuring HTTPRoute request header modifications.

xref: https://github.com/envoyproxy/gateway/issues/501

### CODE REVIEW

**Review by danehans — CHANGES_REQUESTED** [review]
Your first commit may conflict with https://github.com/envoyproxy/gateway/pull/595. Can you pull it out into a separate PR so the HTTP Request Headers doc doesn't get blocked?

File: docs/user/QUICKSTART.md [hunk 1]
```diff
@@ -16,39 +16,43 @@ Install the Gateway API CRDs:
...
 
-Create the HTTPRoute to route traffic through Envoy proxy to the example app:
+   ```shell
+   kubectl apply -f https://raw.githubusercontent.com/envoyproxy/gateway/v0.2.0-rc2/examples/kubernetes/gateway.yaml
+   ```
 
-```shell
-kubectl apply -f https://raw.githubusercontent.com/envoyproxy/gateway/v0.2.0-rc2/examples/kubernetes/httproute.yaml
-```
+   After applying the Gateway manifest above, a corresponding Deployment and Service will be created.
```
[the remark this answers is not in the document]
Comment (AliceProxy) [reply docs/user/QUICKSTART.md]: Done, new PR for the quickstart changes is https://github.com/envoyproxy/gateway/pull/598

File: docs/user/HTTP_REQUEST_HEADERS.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
+# HTTP Request Headers
+
+The [HTTPRoute][] resource can modify the headers of a request before forwarding it to the upstream service. HTTPRoute rules cannot use both filter types at once. Currently, Envoy Gateway only supports __core__
+[HTTPRoute filters][] which consist of `RequestRedirect` and `RequestHeaderModifier` at the time of this writing. To
+learn more about HTTP routing, refer to the [Gateway API documentation][].
+
+A [`RequestHeaderModifier` filter][req_filter] instructs Gateways to modify the headers in requests that match the rule before forwarding the request upstream. Note that the `RequestHeaderModifier` filter will only modify headers before the request is sent from Envoy to the upstream service and will not affect response headers returned to the downstream client.
+
+Follow the steps from the [Secure Gateways](secure-gateways.md) to install Envoy Gateway and the example manifest. Do not
+proceed until you can curl the example backend from the Quickstart guide using HTTPS.
```
Comment (danehans) [inline docs/user/HTTP_REQUEST_HEADERS.md:10]: I don't see why the secure gateways guide is needed for this guide. Can you update to reference the quickstart guide?

File: docs/user/HTTP_REQUEST_HEADERS.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+  - headers.example
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /
+    backendRefs:
+    - group: ""
+      kind: Service
+      name: httpbin
```
Comment (danehans) [inline docs/user/HTTP_REQUEST_HEADERS.md:35]: The quickstart no longer uses httpbin. The name should be `backend` and port `3000`.

xref: https://github.com/envoyproxy/gateway/blob/main/examples/kubernetes/quickstart.yaml


Update throughout this doc.

**Review by danehans — CHANGES_REQUESTED** [review]
A few more things I found that need to be fixed.

**Review by danehans — APPROVED** [review]

**Pull request merged**